### PR TITLE
Extract preprocessing  as separate task

### PR DESCRIPTION
**Pull request by kamilrakoczy**
Extract preprocessing as separate task, add it to JPetManager as 1st task, add goodRun test to JPetManager, fix printing options for task in JPetTaskChainExecutor

### CODE REVIEW

File: JPetPreprocessingTask/JPetPreprocessingTask.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+
+#ifndef JPETPREPROCESSINGTASK_H
+#define JPETPREPROCESSINGTASK_H
+
+#include "../JPetTask/JPetTask.h"
+#include "../JPetParams/JPetParams.h"
+#include "../JPetParamGetterAscii/JPetParamSaverAscii.h"
+
+class JPetPreprocessingTask : public JPetTask
```
Comment (wkrzemien) [inline JPetPreprocessingTask/JPetPreprocessingTask.h]: I think this task should have more meanigful name with relation to what it is actually doing.
JPEtParamBankHandler ? Think about it.
Also, add a short class description that describes what this task really does

File: JPetPreprocessingTask/JPetPreprocessingTask.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  explicit JPetPreprocessingTask(const char* name = "");
+  bool init(const JPetParamsInterface& inOptions) override;
+  bool run(const JPetDataInterface&) override;
+  bool terminate(JPetParamsInterface&) override;
+
+private:
+  JPetParams params;
+};
+
+#endif /*  !JPETPREPROCESSINGTASK_H */
```
Comment (wkrzemien) [inline JPetPreprocessingTask/JPetPreprocessingTask.h]: endline

**Comment by kamilrakoczy** [comment]
I changed name to JPetParamBankHandlerTask, I was also thinking about JPetParamBankFillerTask, if it is better name I can change it again.

File: JPetManager/JPetManager.cpp [hunk 3]
```diff
@@ -113,6 +114,10 @@ bool JPetManager::parseCmdLine(int argc, const char** argv)
       return new JPetUnzipAndUnpackTask("UnpackerAndUnzipper");
     };
     fTaskGeneratorChain->insert(fTaskGeneratorChain->begin(), task);
+    auto preprocessingTask = []() {
+      return new JPetParamBankHandlerTask("Preprocessing");
```
Comment (wkrzemien) [inline JPetManager/JPetManager.cpp]: Change "Preprocessing" to eg. ParamBankFilling

**Review by wkrzemien — APPROVED** [review]
For me everything is fine.

**Comment by wkrzemien** [comment]
@alekgajos pls check that this PR is ok for you

**Comment by kamilrakoczy** [comment]
It is not completed yet.
generateParamBankFromConfig(const JPetParams& params) is not working correctly, but generateParamBankFromRootFile(const JPetParams& params) should work.

**Pull request closed without merging**